### ISSUE TITLE
JWT auth and reservations api

### ISSUE THREAD

**Issue by stefansaenger** [jitsi/jitsi-meet#10947]
### Description:

We are using lib-jitsi-meet in our own application, and our application uses authentication with JWT. When we recently tried to integrate the reservations api as described in https://jitsi.github.io/handbook/docs/devops-guide/reservation we came across this:

REST calls to reservation api would not use mail_owner from JWT data, but only use the actual anonymous jid assigned on connecting.

### Steps to reproduce:

1. set up reservation system and create a reservation for a call, setting a mail_owner 
2. at the scheduled time, try to start teh conference using JWT auth, with user.email in JWT set to the reserved mail_owner

### Expected behavior:

A user that has the correct user.email in his JWT is able to start the conference.

### Actual behavior:

A user that has the correct user.email in his JWT is not able to start the conference; instead he receives a reservation error.

### Server information:

- Jitsi Meet version: jitsi-meet 2.0.6865-2

**Comment by shawnchin** [comment]
Revisited this because I was looking to achieve something similar i.e. embedding additional user context from JWT into payload of `/conference`.

This looks simple on the surface, but in practice gets rather messy when you have multiple people attempting to join around the same time.

The current [implementation](https://jitsi.github.io/handbook/docs/devops-guide/reservation/#implementation-diagram) of mod_reservations fires the POST call when the first room creation request is made, and does not repeat the call for subsequent joins while this request is in flight. Rejection responses are also [cached for a short period of time](https://github.com/jitsi/jitsi-meet/blob/9c83831b2c84d4253968b9815f9f10b247324b2b/resources/prosody-plugins/mod_reservations.lua#L276-L279). This means that when there are multiple users joining within a ~90 seconds window, the outcome for users could be inconsistent if we depend on user payload because the API endpoint will only ever see the payload from the first user.

In short, without changing how mod_reservations work, I don't see how we can reliably make reservations decisions based on user context. Even the existing use of `mail_owner` is suspect since a guest user joining before authentication user would result in both users getting a reservation error response.

@damencho Thoughts? 

I wonder if the deprecated Jicofo version of reservations had the same issue, or if it made POST calls for every join request 🤔

**Comment by damencho** [comment]
> I wonder if the deprecated Jicofo version of reservations had the same issue, or if it made POST calls for every join request 🤔

No idea :) I would guess it sends POST only one time:
https://github.com/jitsi/jicofo/pull/852/files#diff-7e8bc87cbc77ffa4a09bbdcee9a7e1d46cf5c4c797a3fc48b1742373b08a1745L152

**Comment by shawnchin** [comment]
Thanks for link to the diffs. Very useful reference.

It looks like it was making API calls within a synchronized method which only updates `conferenceMap` on successful response, so it is possible it was sequentially making multiple API calls for the same room but stops if there was a successful creation? 

https://github.com/jitsi/jicofo/blob/24b0bbdc854dbb660f655fff86daf80d86d82a8f/src/main/java/org/jitsi/impl/reservation/rest/RESTReservations.java#L114-L127

in which case it won't have the same issue.

Alas, I'm not sure there will be much appetite to rewrite mod_reservations at the moment to deal with this 😮‍💨

**Comment by damencho** [comment]
Yep, seems in sync so only one will create it and for the rest there will be other response ...
